sdk/python/jobs/automl-standalone-jobs/automl-distributed-regression-task-hardware-performance/regression-distributed-lightgbm.ipynb

```diff
@@ -0,0 +1,1505 @@
+    "my_training_data_input = Input(type=AssetTypes.MLTABLE, path=train_mltable_path)\n",
+    "\n",
+    "# WITH REMOTE PATH: If available already in the cloud/workspace-blob-store\n",
+    "# my_training_data_input = Input(type=AssetTypes.MLTABLE, path=\"azureml://datastores/workspaceblobstore/paths/my_training_mltable\")"
+   ],
+   "outputs": [],
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": false,
+     "outputs_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    },
+    "gather": {
+     "logged": 1684768802283
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "For documentation on creating your own MLTable assets for jobs beyond this notebook:\r\n",
+    "- https://learn.microsoft.com/en-us/azure/machine-learning/reference-yaml-mltable details how to write MLTable YAMLs (required for each MLTable asset).\r\n",
+    "- https://learn.microsoft.com/en-us/azure/machine-learning/how-to-create-data-assets?tabs=Python-SDK covers how to work with them in the v2 CLI/SDK."
+   ],
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## 3. Create or Attach existing AmlCompute\r\n",
+    "\r\n",
+    "[Azure Machine Learning Compute](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-set-up-training-targets#amlcompute) is a managed-compute infrastructure that allows the user to easily create a single or multi-node compute. In this tutorial, you will create and an AmlCompute cluster as your training compute resource.\r\n",
+    "\r\n",
+    "<b>Creation of AmlCompute takes approximately 5 minutes.</b>\r\n",
+    "\r\n",
+    "If the AmlCompute with that name is already in your workspace this code will skip the creation process.\r\n",
+    "As with other Azure services, there are limits on certain resources (e.g. AmlCompute) associated with the Azure Machine Learning service. Please read [this article](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-manage-quotas) on the default limits and how to request more quota."
+   ],
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "from azure.core.exceptions import ResourceNotFoundError\n",
+    "from azure.ai.ml.entities import AmlCompute\n",
+    "\n",
+    "cluster_name = \"automl-distributed-cluster\"\n",
+    "\n",
+    "try:\n",
+    "    # Retrieve an already attached Azure Machine Learning Compute.\n",
+    "    compute = ml_client.compute.get(cluster_name)\n",
+    "except ResourceNotFoundError as e:\n",
+    "    compute = AmlCompute(\n",
+    "        name=cluster_name,\n",
+    "        size=\"STANDARD_DS12_V2\",\n",
+    "        type=\"amlcompute\",\n",
+    "        min_instances=4,\n",
```

Comment:
A little dangerous IMO to set `min_instance=4`. Then the cluster is costing money as long as it exists. Might be unexpected for the customer. We usually set this to 0

Reply:
@EricWrightAtWork  This is a distributed training and it is mandatory to have at least 4 instances. So, in case the compute_cluster with given name does not exist, then the creation of new one should have at least 4 instances. That's why I added this. Let me know your thoughts.

Reply:
@Nivi09 I agree with @EricWrightAtWork on this. Make sure to explain this in a paragraph **before** the code block that creates the cluster. Imagine a scenario where a user tries this notebook and creates a 4 node (minimum) cluster. The user runs the notebook and forgets about it. This 4-node cluster will continue running until the user deletes it. By that time, they will accumulate potentially a large bill. This is the reason why we always set min cluster nodes to 0, to avoid scenarios like this. If the user creates a 1, 2, or 3 node cluster and we explained that they need a minimum of 4 nodes. Then this is on the user.

Reply:
Thank you. Added note as below :  
3. Create or Attach existing AmlCompute
[Azure Machine Learning Compute](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-set-up-training-targets#amlcompute) is a managed-compute infrastructure that allows the user to easily create a single or multi-node compute. In this tutorial, you will create and an AmlCompute cluster as your training compute resource. If the AmlCompute with that name is already in your workspace this code will skip the creation process. Else code will create compute resource with given configurations. Please set the parameter min_instances to at least 4 in configuration as AutoML needs at least 4 instances to run distributed training.

Creation of AmlCompute takes approximately 5 minutes.

With other Azure services, there are limits on certain resources (e.g. AmlCompute) associated with the Azure Machine Learning service. Please read [this article](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-manage-quotas) on the default limits and how to request more quota.